jujugui/static/gui/src/app/utils/analytics.js

```diff
@@ -0,0 +1,90 @@
+/*
+This file is part of the Juju GUI, which lets users view and manage Juju
+environments within a graphical interface (https://launchpad.net/juju-gui).
+Copyright (C) 2017 Canonical Ltd.
+
+This program is free software: you can redistribute it and/or modify it under
+the terms of the GNU Affero General Public License version 3, as published by
+the Free Software Foundation.
+
+This program is distributed in the hope that it will be useful, but WITHOUT
+ANY WARRANTY; without even the implied warranties of MERCHANTABILITY,
+SATISFACTORY QUALITY, or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU Affero
+General Public License for more details.
+
+You should have received a copy of the GNU Affero General Public License along
+with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+'use strict';
+
+YUI.add('analytics', function(Y) {
+
+  var ns = Y.namespace('juju');
+
+  ns.analytics = {
```

Comment:
Is there a reason why this is an object and not just a function? Are you already planning to extend the API with other methods? If not, I'd say let's just define it as a function and pass it as "sendAnalytics" to components.